.github/workflows/ci.yml

```diff
@@ -1,23 +1,39 @@
 name: CI
+
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
 jobs:
   test:
+    runs-on: ubuntu-latest
     strategy:
       matrix:
+        python-version: [3.12]
```

Comment:
I'm not sure we want to run CI on a version that we're not running the app on:

https://github.com/cabotage/cabotage-app/blob/6b9b216f045bcaf742b14eca6da11237a19b664a/Dockerfile#L1

I think 3.11 is best for now until we target 3.12 for runtime.